packages/client-config/src/generate_client_config_to_file.ts

```diff
@@ -1,39 +1,60 @@
 import { AwsCredentialIdentityProvider } from '@aws-sdk/types';
 import { generateClientConfig } from './generate_client_config.js';
 import { ClientConfigWriter } from './client-config-writer/client_config_writer.js';
-import { ClientConfigFormat } from './client-config-types/client_config.js';
+import {
+  ClientConfigFormat,
+  ClientConfigVersion,
+  ClientConfigVersions,
+} from './client-config-types/client_config.js';
 import { getClientConfigPath } from './paths/index.js';
 import { DeployedBackendIdentifier } from '@aws-amplify/deployed-backend-client';
-import { ClientConfigFormatter } from './client-config-writer/client_config_formatter.js';
-import { ClientConfigConverter } from './client-config-writer/client_config_converter.js';
+import { ClientConfigMobileConverter } from './client-config-writer/client_config_to_mobile_legacy_converter.js';
 import { fileURLToPath } from 'url';
 import * as fsp from 'fs/promises';
+import { ClientConfigFormatterLegacy } from './client-config-writer/client_config_formatter_legacy.js';
+import { ClientConfigFormatterGen2 } from './client-config-writer/client_config_formatter_gen2.js';
+import { getClientConfigFileName } from './paths/get_client_config_name.js';
 
 /**
  * Main entry point for generating client config and writing to a file
  */
 export const generateClientConfigToFile = async (
   credentialProvider: AwsCredentialIdentityProvider,
   backendIdentifier: DeployedBackendIdentifier,
+  version: ClientConfigVersion,
   outDir?: string,
   format?: ClientConfigFormat,
   // TODO: update this type when Printer interface gets defined in platform-core.
   log?: (message: string) => void
 ): Promise<void> => {
   const packageJson = await readPackageJson();
 
+  const isLegacyConfig = version === ClientConfigVersions.LEGACY;
+
   const clientConfigWriter = new ClientConfigWriter(
     getClientConfigPath,
-    new ClientConfigFormatter(
-      new ClientConfigConverter(packageJson.name, packageJson.version)
-    )
+    isLegacyConfig
+      ? new ClientConfigFormatterLegacy(
+          new ClientConfigMobileConverter(packageJson.name, packageJson.version)
+        )
+      : new ClientConfigFormatterGen2()
   );
 
   const clientConfig = await generateClientConfig(
     credentialProvider,
-    backendIdentifier
+    backendIdentifier,
+    version
+  );
+
+  const fileName = getClientConfigFileName(version);
```

Comment:
this seems like it should be internal to the ClientConfigWriter. Since we are already passing in a different ClientConfigFormatter to ClientConfigWriter based on the version, perhaps the ClientConfigFormatter could be responsible for knowing what filename to write to?